sdk/search/search-documents/src/searchIndexClient.ts

```diff
@@ -425,11 +443,19 @@ export class SearchIndexClient {
     const { span, updatedOptions } = createSpan("SearchIndexClient-deleteSynonymMap", options);
     try {
       const synonymMapName: string = typeof synonymMap === "string" ? synonymMap : synonymMap.name;
+      const etag =
+        typeof synonymMap === "string"
```

Comment:
would something like `const etag = typeof synonymMap === "string" && options.onlyIfUnchanged ? synonymMap.etag : undefined` be simpler?

Reply:
But if synonymmap is a string and if I set onlyIfUnchanged to true, the code will get into trouble. I wanted to avoid that